crates/core/tedge_api/src/alarm.rs

```diff
@@ -145,6 +150,47 @@ impl ThinEdgeAlarm {
             ))
         }
     }
+
+    pub fn new_try_from(
+        parent_device_name: String,
+        mqtt_topic: &str,
+        mqtt_payload: &str,
+    ) -> Result<Self, ThinEdgeJsonDeserializerError> {
+        let topic_split: Vec<&str> = mqtt_topic.split('/').collect();
+
+        let alarm_name = topic_split.last().cloned().unwrap_or_default();
+        if alarm_name.is_empty() {
+            return Err(ThinEdgeJsonDeserializerError::UnsupportedTopic(
+                mqtt_topic.into(),
+            ));
+        }
+        let alarm_data: Option<ThinEdgeAlarmData> = if mqtt_payload.is_empty() {
+            None
+        } else {
+            Some(serde_json::from_str(mqtt_payload)?)
+        };
+
+        let external_source = if topic_split[2].eq("main") {
+            parent_device_name
+        } else {
+            new_get_child_id_from_topic(parent_device_name, mqtt_topic.into()).unwrap_or_default()
+        };
+
+        let severity = match alarm_data.clone() {
+            Some(data) => match data.severity {
+                Some(alarm_severity) => alarm_severity,
+                None => return Err(ThinEdgeJsonDeserializerError::AlarmSeverityNotFound),
+            },
+            None => return Err(ThinEdgeJsonDeserializerError::AlarmSeverityNotFound),
```

Comment:
Using a default severity will be more convenient for users. I would go with using a severity of `major` if it is not provided by the user.

We could even make this configurable later on.

Reply:
done